pkg/apiclient/offline-client.go

```diff
@@ -50,42 +49,43 @@ func newOfflineClient(paths []string) (context.Context, Client, error) {
 
 	for _, basePath := range paths {
 		err := file.WalkManifests(basePath, func(path string, bytes []byte) error {
-			var generic map[string]interface{}
-			if err := yaml.Unmarshal(bytes, &generic); err != nil {
-				return fmt.Errorf("failed to parse YAML from file %s: %w", path, err)
-			}
-			switch generic["kind"] {
-			case "ClusterWorkflowTemplate":
-				cwftmpl := new(wfv1.ClusterWorkflowTemplate)
-				if err := yaml.Unmarshal(bytes, &cwftmpl); err != nil {
-					return fmt.Errorf("failed to unmarshal file %s as a ClusterWorkflowTemplate: %w", path, err)
+			for _, pr := range common.ParseObjects(bytes, false) {
+				obj, err := pr.Object, pr.Err
+				if err != nil {
+					return fmt.Errorf("failed to parse YAML from file %s: %w", path, err)
 				}
 
-				if _, ok := clusterWorkflowTemplateGetter.clusterWorkflowTemplates[cwftmpl.Name]; ok {
-					return fmt.Errorf("duplicate ClusterWorkflowTemplate found: %q", cwftmpl.Name)
+				if obj == nil {
+					continue // could not parse to kubernetes object
 				}
-				clusterWorkflowTemplateGetter.clusterWorkflowTemplates[cwftmpl.Name] = cwftmpl
 
-			case "WorkflowTemplate":
-				wftmpl := new(wfv1.WorkflowTemplate)
-				if err := yaml.Unmarshal(bytes, &wftmpl); err != nil {
-					return fmt.Errorf("failed to unmarshal file %s as a WorkflowTemplate: %w", path, err)
-				}
-				getter, ok := workflowTemplateGetters[wftmpl.Namespace]
-				if !ok {
-					getter = &offlineWorkflowTemplateNamespacedGetter{
-						namespace:         wftmpl.Namespace,
-						workflowTemplates: map[string]*wfv1.WorkflowTemplate{},
+				objName := obj.GetName()
+				namespace := obj.GetNamespace()
```

Comment:
same here that this is copied from `lint.go`, but the naming could be consistent with `objName` as `objNs` in both places
```suggestion
				objNs := obj.GetNamespace()
```